riklet/src/core.rs

```diff
@@ -93,76 +123,276 @@ impl Riklet {
                 self.delete_workload(workload).await?;
             }
             _ => {
-                log::error!("Method not allowed")
+                event!(Level::ERROR, "Method not allowed")
             }
         }
 
         Ok(())
     }
 
+    fn download_image(url: &String, file_path: &String) -> Result<(), Box<dyn Error>> {
+        event!(
+            Level::DEBUG,
+            "Downloading image from {} to {}",
+            url,
+            file_path
+        );
+
+        let mut easy = Easy::new();
+        let mut buffer = Vec::new();
+        easy.url(&url)?;
+        easy.follow_location(true)?;
+
+        {
+            let mut transfer = easy.transfer();
+            transfer.write_function(|data| {
+                buffer.extend_from_slice(data);
+                Ok(data.len())
+            })?;
+            transfer.perform()?;
+        }
+
+        let response_code = easy.response_code()?;
+        if response_code != 200 {
+            return Err(format!("Response code from registry: {}", response_code).into());
+        }
+
+        {
+            event!(Level::DEBUG, "Writing data to {}", file_path);
+            let mut file = File::create(&file_path)?;
+            file.write_all(buffer.as_slice())?;
+        }
+
+        Ok(())
+    }
+
+    fn decompress(source: &Path, destination: &Path) -> Result<(), Box<dyn Error>> {
+        let input_file = File::open(source)?;
+        let mut decoder = Decoder::new(input_file)?;
+        let mut output_file = File::create(destination)?;
+        io::copy(&mut decoder, &mut output_file)?;
+        Ok(())
+    }
+
     async fn create_workload(
         &mut self,
         workload: &InstanceScheduling,
     ) -> Result<(), Box<dyn Error>> {
+        event!(Level::DEBUG, "Creating workload");
         let workload_definition: WorkloadDefinition =
             serde_json::from_str(&workload.definition[..]).unwrap();
         let instance_id: &String = &workload.instance_id;
-        let containers = workload_definition.get_containers(instance_id);
 
-        // Inform the scheduler that the workload is creating
-        self.send_status(5, instance_id).await;
+        if workload_definition.kind == "Function" {
```

Comment:
New workload type should be "Microvm", no "Function".

Reply:
This change is in work in another PR